Automattic/jetpack
# Footer credit: Render default credit on block themes

Part of https://github.com/Automattic/dotcom-forge/issues/8231
Counterpart of D156044-code for Atomic sites

## Proposed changes:

Stops overriding the default footer credit with a custom WP.com footer credit on block themes because such custom credit is unsupported in the site editor.

Before | After
--- | ---
<img width="1122" alt="Screenshot 2024-07-22 at 18 56 36" src="https://github.com/user-attachments/assets/08d0997c-2211-4546-a9cb-524514cdffa4"> | <img width="934" alt="Screenshot 2024-07-23 at 10 07 20" src="https://github.com/user-attachments/assets/29db0690-8882-48ea-81fd-568233bab6eb">


### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Apply these changes to a WoA dev site
- Install Jetpack Beta
- Go to `/wp-admin/admin.php?page=jetpack-beta&plugin=wpcomsh` 
- Activate the branch of this PR
- Make sure your site has a block theme
- Visit the frontend
- Make sure default credit is displayed instead of the custom WP.com credit in the footer
- Open the site editor
- Make some edits to the credit (you can even it remove it if you want)
- Make sure they are reflected in the frontend
- Switch to a classic theme
- Make sure the WP.com custom credit is displayed now
